src/sentry/integrations/jira/utils/api.py

```diff
@@ -52,27 +53,34 @@ def handle_assignee_change(
     issue_key = data["issue"]["key"]
 
     log_context = {"issue_key": issue_key, "integration_id": integration.id}
-    assignee_changed = any(
-        item for item in data["changelog"]["items"] if item["field"] == "assignee"
-    )
-    if not assignee_changed:
-        logger.info("jira.assignee-not-in-changelog", extra=log_context)
-        return
-
-    # If there is no assignee, assume it was unassigned.
-    fields = data["issue"]["fields"]
-    assignee = fields.get("assignee")
-
-    if assignee is None:
-        sync_group_assignee_inbound(integration, None, issue_key, assign=False)
-        return
-
-    email = get_assignee_email(integration, assignee, use_email_scope)
-    if not email:
-        logger.info("jira.missing-assignee-email", extra=log_context)
-        return
 
-    sync_group_assignee_inbound(integration, email, issue_key, assign=True)
+    with ProjectManagementEvent(
+        ProjectManagementActionType.INBOUND_ASSIGNMENT_SYNC, integration=integration
+    ).capture() as lifecycle:
+        assignee_changed = any(
+            item for item in data["changelog"]["items"] if item["field"] == "assignee"
+        )
+        if not assignee_changed:
+            lifecycle.record_halt()
```

Comment:
could this possibly be a success? are these normal scenarios where jira doesn't send us assignee?

Reply:
That's fair. Tbh though, I'm looking at moving this call into the shared util this calls as there are specific lifecycles we probably want to track there, so this is likely to change.